numpy/_core/numeric.py

```diff
@@ -180,6 +201,11 @@ def ones(shape, dtype=None, order='C', *, like=None):
            [1.,  1.]])
 
     """
+    if device not in ["cpu", None]:
```

Comment:
Here again I would suggest to pass on `device` to `empty` and `_ones_with_like`

Reply:
Done!